src/Passes/Source/Generators/IProfileGenerator.cpp

```diff
@@ -0,0 +1,101 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#include "Generators/IProfileGenerator.hpp"
+#include "RuleTransformationPass/Configuration.hpp"
+#include "ValidationPass/ValidationConfiguration.hpp"
+
+#include "Llvm/Llvm.hpp"
+
+namespace microsoft
+{
+namespace quantum
+{
+
+    Profile IProfileGenerator::newProfile(String const& name, OptimizationLevel const& optimisation_level, bool debug)
+    {
+        auto qubit_allocation_manager  = BasicAllocationManager::createNew();
+        auto result_allocation_manager = BasicAllocationManager::createNew();
+
+        auto cfg = configuration_manager_.get<RuleTransformationPassConfiguration>();
+        qubit_allocation_manager->setReuseRegisters(cfg.reuseQubits());
+        result_allocation_manager->setReuseRegisters(cfg.reuseResults());
+
+        // Creating profile
+        // TODO(tfr): Set target machine
+        Profile ret{name, debug, nullptr, qubit_allocation_manager, result_allocation_manager};
+
+        auto module_pass_manager = createGenerationModulePass(ret, optimisation_level, debug);
+
+        ret.setModulePassManager(std::move(module_pass_manager));
+
+        // Creating validator
+        auto validator = std::make_unique<Validator>(configuration_manager_.get<ValidationPassConfiguration>(), debug);
+        ret.setValidator(std::move(validator));
+
+        return ret;
+    }
+
+    llvm::ModulePassManager IProfileGenerator::createGenerationModulePass(
```

Comment:
The function name `createGenerationModulePass()` tells that the function is expected to create some Pass, but the function (creates and) returns some Manager. That's somewhat confusing.

Reply:
Fixed.